keras_nlp/models/whisper/whisper_audio_feature_extractor.py

```diff
@@ -251,3 +256,58 @@ def get_config(self):
             }
         )
         return config
+
+    @classproperty
+    def presets(cls):
+        return copy.deepcopy(backbone_presets)
+
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        **kwargs,
+    ):
+        """Instantiate Whisper audio feature extractor from a given preset.
+
+        Args:
+            preset: string. Must be one of "{{preset_names}}".
+
+        Examples:
+        ```python
+        # Load a preset tokenizer.
+        audio_feature_extractor = WhisperAudioFeatureExtractor.from_preset(
+            "{{example_preset_name}}"
+        )
+
+        # Compute the log-mel spectrogram.
+        audio_tensor = tf.ones((8000,), dtype=tf.float32)
+        audio_feature_extractor(audio_tensor)
+        ```
+        """
+
+        if not cls.presets:
+            raise NotImplementedError(
+                "No presets have been created for this class"
+            )
+
+        if preset not in cls.presets:
+            raise ValueError(
+                "`preset` must be one of "
+                f"""{", ".join(cls.presets)}. Received: {preset}."""
+            )
+
+        config = cls.presets[preset]["audio_feature_extractor_config"]
+
+        # Override the config if any field is present in kwargs. Delete the
+        # field from kwargs.
+        for key in config:
```

Comment:
do we really need this? we don't seem to have it elsewhere. I think kwargs will override config in the line below with this block

Reply:
I believe this line

```
 cls.from_config({**config, **kwargs})
```

 will error out if there are common keys in `config` and `kwargs`. Hence, this snippet.

Reply:
It shouldn't.

```python
one = {"foo": 2}
two = {"foo": 3, "bar": 3}
{**one, **two}
> {'foo': 3, 'bar': 3}
```

Reply:
Ah, my bad. `{**config, **kwargs}` is inside a dict, lol